src/global_components/searchbar.jsx

```diff
@@ -0,0 +1,103 @@
+    }
+
+    drawSearchResults() {
+        let searchResults = [];
+        if (this.props.results){
+            searchResults = this.props.results.map(course => {
+                return (<Dropdown.Item key={course.id.toString()} href={'/course/' + course.id}>{course.title}</Dropdown.Item>);
+            });
+        }
+        const Results = styled(Dropdown.Menu)`
+            position: absolute;
+            width: 100%;
+            margin-top: 0px;
+            :empty {
+                display: none;
+            }
+        `;
+
+        return(
+            <Results show>
+                {searchResults}
+            </Results>
+        );
+    }
+
+    render() {
+        const logoUrl = process.env.PUBLIC_URL + '/logo.gif';
+        return (
+            <div>
+                <Navbar bg='light' variant='light' className='justify-content-around' >
+                    <Navbar.Brand href='/'>
+                        <img
+                            src={logoUrl}
+                            width='30'
+                            height='30'
+                            className='d-inline-block align-top'
+                            alt='React Bootstrap logo'
+                        />
+                    </Navbar.Brand>
+                    <MediaQuery minDeviceWidth={992}>
+                        <Form className='position-relative' inline>
+                            <FormControl type='text' placeholder='Search for a class' className='w-100' onChange={this.requestSearchResults}/>
+                            {this.drawSearchResults()}
+                        </Form>
+                        <LoginButton/>
+                    </MediaQuery>
+                    <MediaQuery maxDeviceWidth={991}>
+                        <Button
+                            className='navbar-toggler-icon px-4 py-4'
+                            onClick={() => this.setState({navOpen: !this.state.navOpen})}
+                            aria-controls='collapsedNavbarContent' 
+                            aria-expanded={this.state.navOpen}
+                        />
+                    </MediaQuery>
+                </Navbar>
+                <MediaQuery maxDeviceWidth={991}>
+                    <Collapse in={this.state.navOpen}>
+                        <div className='bg-light' id='collapsedNavbarContent'>
+                            <Form className='position-relative' inline>
+                                <FormControl type='text' placeholder='Search for a class' className='w-100 mx-2 my-2' onChange={this.requestSearchResults} />
+                                {this.drawSearchResults()}
+                            </Form>
+                            <LoginButton/>
+                        </div>
+                    </Collapse>
+                </MediaQuery>
+            </div>
+            );
+    }
+}
+const mapStateToProps = state => ({
+    results: (state.searchReducers.getSearchResults.results)
```

Comment:
```suggestion
    results: (state.searchReducers.getSearchResults.results) || [] 
```
to avoid having to do undefined checks